pkg/conversion/mapi2capi/machineset.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapi2capi
+
+import (
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// fromMAPIMachineSetToCAPIMachineSet takes a MAPI MachineSet and returns a converted CAPI MachineSet.
+func fromMAPIMachineSetToCAPIMachineSet(mapiMachineSet *mapiv1.MachineSet) (*capiv1.MachineSet, utilerrors.Aggregate) {
```

Comment:
I notice `MachineSet`s don't have a helper to check the `ObjectMeta` information. Should they?

Reply:
Which ObjectMeta are you talking about? There are three levels of ObjectMeta in the MachineSet and it's only the one in spec.template.spec.metadata that needs checks, but that comes from the machine conversion level if you look at where this is called.